### ISSUE TITLE
error: adding 'int' to a string does not append to the string

### ISSUE THREAD

**Issue by darkoverlordofdata** [ObjFW/ObjFW#72]
While following the instructions starting at _Getting, building and installing ObjFW_ 

First I get a problem trying to clone from git:

darko@DESKTOP-DOJU898 MINGW64 /c/Users/darko/Documents/GitHub
$ git clone https://heap.zone/git/objfw.git
Cloning into 'objfw'...
error: inflate: data stream error (incorrect header check)
error: inflate: data stream error (incorrect header check)
error: inflate: data stream error (incorrect header check)
error: inflate: data stream error (incorrect header check)
error: inflate: data stream error (incorrect header check)
error: inflate: data stream error (incorrect header check)
error: File 1784294c3c48dc7edc631b7237dd928bfe343d22 (https://heap.zone/git/objfw.git/objects/f8/b804d057d29564d84b02523069a0ea691c8238) corrupt
error: Unable to find f8b804d057d29564d84b02523069a0ea691c8238 under https://heap.zone/git/objfw.git
Cannot obtain needed blob f8b804d057d29564d84b02523069a0ea691c8238
while processing commit 130ceda651aa637becc22067d45216c97dc7c879.
error: fetch failed.

Next, I was hoping that this version https://github.com/ObjFW/ObjFW.git  would work - it clones ok, but gets a compile error:

darko@DESKTOP-DOJU898 MINGW64 /c/Users/darko/Documents/GitHub/ObjFw
$ ./autogen.sh && ./configure && make -j16 install
configure: loading site script /mingw64/etc/config.site
...
`Successfully generated dependencies.
OFStringTests.m:359:59: error: adding 'int' to a string does not append to the string [-Werror,-Wstring-plus-int]
            R([s[0] appendUTF8String: "foo\xEF\xBB\xBF" "barqux" + 3
                                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~
./TestsAppDelegate.h:56:17: note: expanded from macro 'R'
#define R(...) (__VA_ARGS__, 1)
                ^~~~~~~~~~~
./TestsAppDelegate.h:25:7: note: expanded from macro 'TEST'
                if (__VA_ARGS__)                        \
                    ^~~~~~~~~~~
OFStringTests.m:359:59: note: use array indexing to silence this warning
            R([s[0] appendUTF8String: "foo\xEF\xBB\xBF" "barqux" + 3
                                                                 ^
                                      &                          [
./TestsAppDelegate.h:56:17: note: expanded from macro 'R'
#define R(...) (__VA_ARGS__, 1)
                ^
./TestsAppDelegate.h:25:7: note: expanded from macro 'TEST'
                if (__VA_ARGS__)                        \
                    ^
1 error generated.
`

**Comment by darkoverlordofdata** [comment]
So, if I comment the test/OFStringTests.m lines 358-367, basically just removing that one test, then everything compiles, and I get a working installation!

**Comment by Midar** [comment]
Thanks, that's a new warning introduced in Clang, apparently. The + 3 was a hack to prevent compilers from becoming smarter and stripping the BOM. Fixed in 2c40e67b.
